command-line/lib/Command/Script/Validation.hs

```diff
@@ -0,0 +1,94 @@
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE OverloadedStrings #-}
+
+{-# OPTIONS_HADDOCK hide #-}
+
+module Command.Script.Validation
+    ( Cmd
+    , mod
+    , run
+
+    ) where
+
+import Prelude hiding
+    ( mod )
+
+import Cardano.Address.Script
+    ( KeyHash
+    , Script (..)
+    , TxValidity (..)
+    , ValidationLevel (..)
+    , prettyErrValidateScript
+    , validateScript
+    )
+import Data.Word
+    ( Word64 )
+import Options.Applicative
+    ( CommandFields
+    , Mod
+    , command
+    , footerDoc
+    , header
+    , helper
+    , info
+    , optional
+    , progDesc
+    )
+import Options.Applicative.Help.Pretty
+    ( bold, indent, string, vsep )
+import Options.Applicative.Script
+    ( levelOpt, scriptArg, txValidOpt )
+import System.IO
+    ( stderr, stdout )
+import System.IO.Extra
+    ( hPutString, progName )
+
+data Cmd = Cmd
+    { script :: Script KeyHash
+    , validationLevel :: ValidationLevel
+    , txValidFrom :: Maybe Word64
+    , txValidTo :: Maybe Word64
+    } deriving (Show)
+
+mod :: (Cmd -> parent) -> Mod CommandFields parent
+mod liftCmd = command "validate" $
+    info (helper <*> fmap liftCmd parser) $ mempty
+        <> progDesc "Validate a script"
+        <> header "Choose a required or recommended validation of a script for a given transaction validation interval."
+        <> footerDoc (Just $ vsep
+            [ string "The script is taken as argument. To have required validation pass '--required'."
+            , string "To have recommended validation pass '--recommended'. Recommended validation includes also"
```

Comment:
It'd be good to explain what `recommended` adds compared to `required`.

Reply:
added explanation plus make --required default and whole option optional, see HEAD